libraries/stdlib/samples/test/samples/collections/arrays.kt

```diff
@@ -64,6 +64,176 @@ class Arrays {
 
     class Transformations {
 
+        @Sample
+        fun associate() {
+            data class Person(val firstName: String, val lastName: String)
+
+            val computerScientists = arrayOf(Person("Grace", "Hopper"), Person("Edsger", "Dijkstra"), Person("Alan", "Turing"))
+
+            val byLastName = computerScientists.associate { it.lastName to it.firstName }
+
+            assertPrints(byLastName, "{Hopper=Grace, Dijkstra=Edsger, Turing=Alan}")
+        }
+
+        @Sample
+        fun associateArrayOfPrimitives() {
+            fun fib(n: Int): Int {
+                var current = 0
+                var next = 1
+                var sum: Int
+                if (n == 0)
+                    return current
+                for (i in 2..n) {
+                    sum = current + next
+                    current = next
+                    next = sum
+                }
+                return next
+            }
```

Comment:
This implementation of the Fibonacci numbers has three vars and three branches. How about the following?
```
fun fib(n: Int): Int =
    when (n) {
        0 -> 0
        1 -> 1
        else -> fib(n - 1) + fib(n - 2)
    }
```
I know it is less efficient, but its much clearer.

Reply:
I went with a simpler example on these instead.